draft-ietf-protected-audience-key-value-services.md

```diff
@@ -0,0 +1,518 @@
+-   Each partition contains one or more key groups. Each key group has its unique attributes among
+    all key groups in the partition. The attributes are represented by a list of `tags`. Besides
+    tags, the key group contains a list of keys to look up.
+-   Each partition has a unique id.
+-   Each partition has a compression group field. Results of partitions belonging to the same
+    compression group can be compressed together in the response. Different compression groups must
+    be compressed separately (see {{compression-group}}).
+
+### Encryption {#request-encryption}
+
+The request is encrypted with [HPKE]. The request uses a repurposed [OHTTP] encapsulation format
+(see {{Section 4.3 of OHTTP}} and {{Section 4.6 of OHTTP}}) with a
+`message/ad-auction-trusted-signals-request` media type instead of `message/bhttp request`.
+
+### Request Schema {#request-schema}
+
+The request is a [CBOR] encoded message with the following [CDDL] schema:
+
+~~~~~ cddl
+request = {
+    ? acceptCompression: [* tstr],
+    ; A list of supported response compression algorithms; must contain at least one of "none", "gzip", "brotli"
+    partitions: [* partition],
+    ; A list of partitions. Each must be processed independently. Accessible by UDF.
+}
+
+partition = {
+    id: uint,
+    ; Unique id of the partition in this request. Used by responses to refer to request partitions.
+    compressionGroupId: uint,
+    ; Unique id of a compression group in this request. Only partitions belonging to the same compression group will be compressed together in the response
+    ? metadata: partitionMetadata,
+    arguments: [* requestArgument],
+    ; One group of keys and common attributes about them
+}
+;Single partition object. A collection of keys that can be processed together.
+
+
+partitionMetadata = {
+    ? hostname: tstr,
+    ; The hostname of the top-level frame calling runAdAuction
+    ? experimentGroupId: tstr,
+    ? slotSize: tstr,
+    ; Available if trustedBiddingSignalsSlotSizeMode=slot-size. In the form of <width>,<height>
+    ? allSlotsRequestedSizes: tstr,
+    ; Available if trustedBiddingSignalsSlotSizeMode=all-slots-requested-sizes. In the form of <width1>,<height1>,<width2>,<height2>,...
+}
+
+requestArgument = {
+    ? tags: [* tstr],
+    ; List of tags describing this group's attributes
+    ? data: [* tstr],
+    ; List of keys to get values for
+}
+~~~~~
+
+#### Available Tags
+
+Each key group has exactly one tag from the `namespace` category.
+
+| Tag | Description |
+|---|---|
+| interestGroupNames | Names of interest groups in the encompassing partition. |
+| keys | “keys” is a list of trustedBiddingSignalsKeys strings. |
+| renderUrls | Similarly, sellers may want to fetch information about a specific creative, e.g. the results of some out-of-band ad scanning system. This works in much the same way, with the base URL coming from the trustedScoringSignalsUrl property of the seller's auction configuration object.  |
```

Comment:
I don't know what "Similarly" is referring similarity to.
It's unclear how "sellers may want to fetch information about a specific creative" relates to renderUrls.
trustedScoringSignalsUrl needs a definiition, which is going to be very complicated.  Perhaps its better to give a more general description like `"renderUrls" represent URLs for advertisements that are used to look up values in a table`